src/main/anovos/drift/detector.py

```diff
@@ -187,55 +254,113 @@ def statistics(
+                .withColumn("attribute", F.lit(str(i)))
+                .select("attribute", "PSI")
+            )
+            temp_method_join_list.append(xy_psi)
+
+        if "HD" in method_type:
+            xy_hd = (
+                xy.withColumn(
+                    "pow",
+                    F.pow((F.sqrt(F.col("p")) - F.sqrt(F.col("q"))), 2),
+                )
+                .select(F.sqrt(F.sum(F.col("pow")) / 2).alias("HD"))
+                .withColumn("attribute", F.lit(str(i)))
+                .select("attribute", "HD")
+            )
+            temp_method_join_list.append(xy_hd)
+
+        if "JSD" in method_type:
+            xy_jsd = (
+                xy.withColumn("m", ((F.col("p") + F.col("q")) / 2))
+                .withColumn("log_pm", (F.col("p") * F.log(F.col("p") / F.col("m"))))
+                .withColumn("log_qm", (F.col("q") * F.log(F.col("q") / F.col("m"))))
+                .select(
+                    F.sum(F.col("log_pm")).alias("pm"),
+                    F.sum(F.col("log_qm")).alias("qm"),
+                )
+                .select(((F.col("pm") + F.col("qm")) / 2).alias("JSD"))
+                .withColumn("attribute", F.lit(str(i)))
+                .select("attribute", "JSD")
+            )
+            temp_method_join_list.append(xy_jsd)
+
+        if "KS" in method_type:
+            xy_ks = (
+                xy.withColumn(
+                    "cum_sum_p",
+                    F.sum(F.col("p")).over(
+                        Window.partitionBy().orderBy().rowsBetween(-sys.maxsize, 0)
+                    ),
+                )
+                .withColumn(
+                    "cum_sum_q",
+                    F.sum(F.col("q")).over(
+                        Window.partitionBy().orderBy().rowsBetween(-sys.maxsize, 0)
+                    ),
+                )
+                .withColumn(
+                    "deduct_abs", F.abs(F.col("cum_sum_p") - F.col("cum_sum_q"))
+                )
+                .select(
+                    F.max(F.col("deduct_abs")).alias("KS"),
+                )
+                .withColumn("attribute", F.lit(str(i)))
+                .select("attribute", "KS")
+            )
+            temp_method_join_list.append(xy_ks)
+
+        xy_temp = temp_method_join_list[0]
+        if len(temp_method_join_list) > 1:
+            for count in range(1, len(temp_method_join_list)):
+                xy_temp = xy_temp.join(
+                    temp_method_join_list[count], "attribute", "inner"
+                )
+
+        temp_list.append(xy_temp)
+
+    def unionAll(dfs):
+        first, *_ = dfs
+        return first.sql_ctx.createDataFrame(
+            first.sql_ctx._sc.union([df.rdd for df in dfs]), first.schema
         )
-        .select(["attribute"] + method_type + ["flagged"])
-        .orderBy(F.desc("flagged"))
+
+    odf_union = unionAll(temp_list)
+    odf = odf_union.withColumn(
+        "flagged",
+        F.coalesce(
```

Comment:
`F.coalesce` returns the first column that is not null so if the value can only be 0 or 1, the first value will always be returned.

Reply:
Thank @sinuochen . I have addressed it in https://github.com/anovos/anovos/pull/306/commits/f5b6c7b9a9219aee958509a853b2476e2116c953. Can you have a look and see if this is resolved or not.